test/rich-text/plugins/link-paste-handler.test.ts

```diff
@@ -0,0 +1,145 @@
+import { linkPasteHandler } from "../../../src/rich-text/plugins/link-paste-handler";
+import "../../matchers";
+import {
+    applySelection,
+    cleanupPasteSupport,
+    createState,
+    createView,
+    dispatchPasteEvent,
+    setupPasteSupport,
+} from "../test-helpers";
+
+const nonURLTestData = [
+    "not a URL",
+    "https://example.org but not by itself",
+    "URL https://example.org in the middle",
+];
+
+const URLTestData = [
+    "https://example.org",
+    "https://example.org/wiki/BestSubjectEver",
+    "https://sub.complicated.domain:8080/path/to/whatever.png#hash?query=parameter%20test",
+];
+
+describe("linkPasteHandler plugin", () => {
+    beforeAll(setupPasteSupport);
+    afterAll(cleanupPasteSupport);
+
+    it.each(nonURLTestData)(
+        "should handle pasting non-URL text (%#)",
+        (text) => {
+            const view = createView(createState("", [linkPasteHandler]));
+
+            dispatchPasteEvent(view.dom, {
+                "text/plain": text,
+            });
+
+            const insertedNode = view.state.doc.nodeAt(
+                view.state.selection.from - 1
+            );
+
+            expect(insertedNode.isText).toBe(true);
+            expect(insertedNode.text).toBe(text);
+            expect(insertedNode.marks).toHaveLength(0); // not a link
+        }
+    );
+
+    it.each(URLTestData)(
+        "should handle pasting URL text without existing selection (%#)",
+        (text) => {
+            const view = createView(createState("", [linkPasteHandler]));
+
+            dispatchPasteEvent(view.dom, {
+                "text/plain": text,
+            });
+
+            const insertedNode = view.state.doc.nodeAt(
+                view.state.selection.from - 1
+            );
+
+            expect(insertedNode.isText).toBe(true);
+            expect(insertedNode.text).toBe(text);
+            expect(insertedNode.marks).toHaveLength(1);
+            expect(insertedNode.marks[0].type.name).toBe("link");
+            expect(insertedNode.marks[0].attrs.href).toBe(text);
+            expect(insertedNode.marks[0].attrs.markup).toBe("linkify");
+            expect(insertedNode.marks[0].attrs.title).toBeNull();
+        }
+    );
+
+    it.each(URLTestData)(
+        "should use existing selection as link text (%#)",
+        (text) => {
+            let state = createState("<p>my example link</p>", [
```

Comment:
nit: I'd like a wider variety of tests for this case, if possible. Things like pasting across newlines, pasting with multiple node types selected (e.g. h1 + p), selections with invalid marks/nodes inside of them, etc.

This is not a blocking issue, but let's add a `//TODO` if you decide not to add the additional tests.